from_dict/_caching.py

```diff
@@ -0,0 +1,27 @@
+import functools
+from typing import Optional
+
+from . import _from_dict
+
+_original_function = _from_dict._get_constructor_type_hints
+_cached_function: Optional['functools._lru_cache_wrapper'] = None
+
+def cache_enable(max_size: int = 100):
+    global _cached_function
+    if _cached_function:
+        if _cached_function.cache_info().maxsize == max_size:
+            return
+        _cached_function.cache_clear()
+    _cached_function = functools.lru_cache(max_size)(_original_function)
+    _from_dict._get_constructor_type_hints = _cached_function
```

Comment:
I think we should make caching a first-class function of `_get_constructor_type_hints` instead of monkey-patching it in from the outside. This feels kind of wrong.